backend/app/Http/Controllers/StudentController.php

```diff
@@ -66,4 +89,20 @@ public function logout()
 
     return response($response, 200);
   }
+
+  public function update(Request $request) {
+    $studentDetails = Auth::user();
+    $student = Student::find($studentDetails->id);
+
+    $student->name = $request->name;
+    $student->username = $request->username;
+    $student->email = $request->email;
+    $student->password = bcrypt($request->password);
+    $student->thumbnail = request()->file('thumbnail')->store('thumbnails');
+
+
+    $student->save();
+
+    return response('Updated successfully', 200);
```

Comment:
There is a shorter way of doing this which is using update() method instead of save so it would look something like this.
```
public function update(Request $request)
{
    $student = Student::find(auth()->user()->id); // or Auth::user()->id (up to you, but do stick to one format for uniformity)

    $student->update([
        'name' => $request->name,
        'username' => $request->username,
        'email' => $request->email,
        'password' => bcrypt($request->password),
        'thumbnail' => request()->file('thumbnail')->store('thumbnails'),
    ]);

    return response('Updated successfully', 200);
}
```

Reply:
Noted Sir. I'll update this one.